src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java

```diff
@@ -161,10 +164,17 @@ private boolean doesUserHaveDepartment(User user) {
         return user.getDepartment() != null;
     }
 
-    public User deleteUser(Integer id) {
-        User user = userRepository.findById(id).orElseThrow(() -> new UserNotFoundException("User not found"));
-        userRepository.delete(user);
-        return user;
+    @Transactional
+    public void deleteUser(Integer id, String jwt) {
+        final User user = userRepository.findById(id).orElseThrow(() -> new UserNotFoundException("user with id: " + id + "not found"));
+        webClientBuilder.build()
+                .delete()
+                .uri(adminBackend + "/users/delete/" + (user.hasSub() ? user.getSub() : "") + (user.hasColaSub() ? "?colaSub=" + user.getColaSub() : ""))
+                .header("Authorization", "Bearer " + jwt)
+                .retrieve()
+                .bodyToMono(Void.class)
```

Comment:
Thinking we should check the response status and cancel the transaction if unsuccessful?

Reply:
bodyToMono will throw a `WebClientException` if there's a 400 or 500 code so that should be fine